### PR TITLE
Nnguide

### PR DESCRIPTION
**Pull request by jungyg**
1. We have implemented the ICCV23 paper "Nearest Neighbor Guidance for Out-of-Distribution Detection" (NNGuide) and appended it to the README File methods list.
2. In order to show the best performance of NNGuide which is dependent to the backbone architecture, we added the files for loading torchvision library's RegNet.
3. In addition, we added the post-processor and configuration file for NNGuide.
4. NNGuide depends on random samples of training dataset using them as bank sets, so we changed the shuffle argument in training data loader in "eval_ood_imagenet.py" to True.
5. By running NNGuide with the default config file(which we committed), results are shown as below.
OOD ::
[nnguide.csv](https://github.com/Jingkang50/OpenOOD/files/13045083/nnguide.csv)

FSOOD:
[nnguide.csv](https://github.com/Jingkang50/OpenOOD/files/13045088/nnguide.csv)

Reference : Park, Jaewoo, Yoon Gyo Jung, and Andrew Beng Jin Teoh. "Nearest Neighbor Guidance for Out-of-Distribution Detection." Proceedings of the IEEE/CVF International Conference on Computer Vision. 2023.

### CODE REVIEW

**Comment by zjysteven** [comment]
Hi @jungyg congrats on your ICCV paper, and thank you for these detailed considerations of integrating NNGuide into OpenOOD, which significantly reduce our burden 👍 We will take a brief overview of your commits asap.

**Comment by zjysteven** [comment]
The NNGuide entry has been added to ImageNet-1K (standard and full-spectrum) leaderboard. Check it [here](https://zjysteven.github.io/OpenOOD/#div_imagenet1k_heading).

**Comment by jungyg** [comment]
Thank you very much for your quick process!

**Comment by jS5t3r** [comment]
How to start NNGuide?

Is there a script? https://github.com/Jingkang50/OpenOOD/tree/main/scripts/ood

**Comment by zjysteven** [comment]
@jS5t3r From the commits of this pull request, it seems the authors of NNGuide only enables the usage via https://github.com/Jingkang50/OpenOOD/blob/main/scripts/eval_ood_imagenet.py, where you can specify the post-processor to be `nnguide`.

**Comment by jS5t3r** [comment]
@zjysteven no result folder iscreated, how to fix this?
